VideoLocker/src/main/java/org/edx/mobile/view/CourseDiscussionResponsesFragment.java

```diff
@@ -102,9 +102,14 @@ public void onDestroy() {
     public void onEventMainThread(DiscussionCommentPostedEvent event) {
         if (discussionThread.containsComment(event.getComment())) {
             discussionThread.incrementCommentCount();
-            courseDiscussionResponsesAdapter.setDiscussionThread(discussionThread);
+            if (event.getParent() == null) {
+                // We got a response
+                courseDiscussionResponsesAdapter.addNewResponse(event.getComment());
+            } else {
+                // We got a comment to a response
+                courseDiscussionResponsesAdapter.addNewComment(event.getParent());
+            }
```

Comment:
>you could simply register for these events from the adapter itself, and perform the necessary checks and additions in there.
I don't like the sound of that approach. The adapter is contained in the fragment and its the fragments duty to change add/update data for the adapter, the adapter is just there to show what its being sent, not manipulate the data on its own.